### ISSUE TITLE
Cannot run individual instrumented tests with method parameters

### ISSUE THREAD

**Issue by matejdro** [mannodermaus/android-junit5#199]
When attempting to run individual instrumented method that has parameters, IDE will just error out with "No tests found"

Example such test: https://github.com/mannodermaus/android-junit5/blob/a2f903443d7c0e48ab4fbbc0951e7cf96e69fc08/instrumentation/sample/src/androidTest/kotlin/de/mannodermaus/sample/ActivityOneTest.kt#L30

**Comment by ktkopone** [comment]
I'm not familiar with the syntax being used in that example file to have parameters in an @Test, but if you're referring to running an individual @ParameterizedTest, I cannot reproduce your issue. I'm able to successfully run individual parameterized instrumented tests via either command line or Android Studio using version 1.2.0 of this repo.

**Comment by matejdro** [comment]
Here is example test:

```
class TestClass {
    @Test
    internal fun testMethod(@TempDir tempDir: File) {

    }
}
```

I've also noticed that I cannot reproduce it every time.

**Comment by mannodermaus** [comment]
As mentioned in the referenced issue, ADB and/or Android Studio seem to discard method signatures with parameters when submitting singular `@Test` runs to the instrumentation. The custom JUnit 5 runner is created properly, but its `run()` method is never invoked by the instrumentation. Looks like we'll have to sit tight until support for `@Test` methods with parameters is added.

**Comment by mannodermaus** [comment]
I was able to conclude a deep and long investigation (don't worry though, it didn't take 1.5 years!). My previous assumption turned out to be semi-correct: The instrumentation discards any test method with parameters because its name doesn't match AndroidX's internal method filter. It works for parameterized tests because we happen to trigger an exemption from a different piece of the AndroidX code, wherein numerical suffixes are understood by the filters. This means that `parameterizedTestMethod[0]` is understood, but `testMethod[String,String]` is not.

I'll try to tackle this for the upcoming 1.3.0 release of the instrumentation runner library. A note for my future self: `TestRequestBuilder$MethodFilter#evaluateTest()` needs to match the test method name exactly, including round brackets and param types!

**Comment by mannodermaus** [comment]
Released in instrumentation 1.3.0! Please note that the new version requires JUnit 5.8.0
